### PR TITLE
Implement request timeouts in unittests

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
Blocked by dependencies. The support code in the node is pretty simple though. And a unittest is included (which passes with all dependencies).

Requires:

- https://github.com/bpfkorea/agora/pull/106

The vibe.d support has not been implemented yet.

### CODE REVIEW

**Comment by AndrejMitrovic** [comment]
Unblocked.

File: source/agora/test/NetworkClient.d [hunk 1]
```diff
@@ -66,3 +66,49 @@ unittest
 
     assert(0, "Nodes should have same block height");
 }
+
+/// test request timeouts
+unittest
+{
+    import vibe.core.log;
+    import core.thread;
```
Comment (Geod24) [inline source/agora/test/NetworkClient.d]: How dare you

Comment (AndrejMitrovic) [reply source/agora/test/NetworkClient.d]: what do you mean?

**Comment by AndrejMitrovic** [comment]
Ugh.. failures.

**Comment by AndrejMitrovic** [comment]
Failure was unrelated, some package failed to download.

**Comment by AndrejMitrovic** [comment]
It passes now.

**Comment by AndrejMitrovic** [comment]
> @Geod24 Geod24 changed the title Implement request timeouts Implement request timeouts in unittests yesterday

Sadness. Anyway, ready to merge?

**Comment by AndrejMitrovic** [comment]
Ugh https://travis-ci.com/bpfkorea/agora/jobs/220751056

**Comment by AndrejMitrovic** [comment]
Ah I think I know what's wrong. It's the logic in my code. I have a `retry_delay` + `max_retries`. So if the `retry_delay` is 100 msecs, and `max_retries` is 20, it should mean after 2 seconds the retrying should stop.

But it's not taking into account the time it takes for the request to time-out. So the actual time spent is:

~retry_delay * max_retries * timeout~

`max_retries * (retry_delay + timeout)`

I need to take this into account.. at least in my unittests.

**Comment by AndrejMitrovic** [comment]
It seems like Travis does some kind of dependency caching.. because it's not finding the localrest commit in my branch, even though it's there (I did update .gitmodules too).

**Comment by AndrejMitrovic** [comment]
Ping.

File: source/agora/test/NetworkClient.d (hunk 1 above)
Comment (Geod24) [reply source/agora/test/NetworkClient.d]: I meant the ordering. But actually the `vibe` import is a debug one, right ?

Comment (AndrejMitrovic) [reply source/agora/test/NetworkClient.d]: I've always ordered them like this:

- local imports first
- dependencies second
- core/std imports after

Comment (AndrejMitrovic) [reply source/agora/test/NetworkClient.d]: Ah the style guide says
> Imports are organized by packages: agora first, dependencies second, std third, core last.

Ok, I will move the core import last.

**Comment by AndrejMitrovic** [comment]
Re-ordered imports.

**Comment by AndrejMitrovic** [comment]
Fixed.

**Review by Geod24 — APPROVED** [review]

**Comment by Geod24** [comment]
Still red

**Comment by AndrejMitrovic** [comment]
Great. Can we drop Mac support?

**Pull request merged**